### ISSUE TITLE
Add getTypes route.

### ISSUE THREAD

**Issue by Nightwing52** [DartExplore/dallas-by-metro-backend#185]
In the current form I'm allowing filtering by type:

![image](https://github.com/DartExplore/dallas-by-metro-backend/assets/18405628/42dfb020-abff-44a6-9099-a26d65bedf8f)

I would like to have a route to get all the types. It would also be nice to add some bean validation to make all types have the following format: RESTAURANT, BAR, COFFEE_SHOP, etc. This can be done with the @ Pattern annotation. Thoughts @ZaneBartlett1?

**Comment by ZaneBartlett1** [comment]
I think having validation like this is good, but for adding an API for it, is there going to be a case where you would need that, but not also be loading a list of POI? The route that serves the POI should have the Type included. Also makes it impossible to do something like present a type that we possibly have in the database but don't have POI for yet. In a case like that we would probably not want to present that type.

**Comment by Nightwing52** [comment]
Yes, I need to load these types on page load so I can display them as options in the form. Does that make sense? Because otherwise I don't have a dynamic list of all the types. Just the ones I manually entered.

For the last issue that actually won't occur. Since it is going to pull the types from the POI table it has to exist in the DB.

**Comment by ZaneBartlett1** [comment]
I guess I'm thinking of when this is in on it's own page, it's going to need to grab the actual list of POI to present on page load, and I figured it would get grabbed from that. I just now remembered at the bottom of the page that won't happen, so having to hit the POI route for just the types, when you're not actually displaying the POI, is a bit much. So sure, we could add that.

Good point that with how types work it necessitates having a POI for it. Kind of odd that we didn't think to denormalize that as well lol
